Doesn't Work With Namespaced Datastores

Wrapping the datastore in a namespace wrapper causes Queries to fail, as per the following output from using `go-ds-bbolt` with `krab`

```
=== RUN   TestKrab
--- FAIL: TestKrab (0.15s)
panic: expected prefix not found [recovered]
        panic: expected prefix not found

goroutine 21 [running]:
testing.tRunner.func1(0xc000152100)
        /usr/local/go/src/testing/testing.go:830 +0x69d
panic(0xa84fc0, 0xd03640)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/ipfs/go-datastore/keytransform.PrefixTransform.InvertKey(0xb2cfd1, 0xd, 0xc00016c0a0, 0x97, 0x0, 0x0)
        /home/solidity/go/pkg/mod/github.com/ipfs/go-datastore@v0.0.5/keytransform/transforms.go:42 +0x229
github.com/ipfs/go-datastore/keytransform.(*Datastore).Query.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/solidity/go/pkg/mod/github.com/ipfs/go-datastore@v0.0.5/keytransform/keytransform.go:78 +0x16c
github.com/ipfs/go-datastore/query.(*resultsIter).NextSync(0xc0000ecaf0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/solidity/go/pkg/mod/github.com/ipfs/go-datastore@v0.0.5/query/query.go:319 +0xd4
github.com/ipfs/go-datastore/query.(*resultsIter).Rest(0xc0000ecaf0, 0x1, 0x1, 0x0, 0x0, 0x0)
        /home/solidity/go/pkg/mod/github.com/ipfs/go-datastore@v0.0.5/query/query.go:330 +0x7f
github.com/RTradeLtd/krab.(*Keystore).List(0xc0000b8ae0, 0xb33011, 0x19, 0xd17180, 0xc0000b8e00, 0x0)
        /home/solidity/Code/krab/krab.go:107 +0x149
github.com/RTradeLtd/krab_test.TestKrab(0xc000152100)
        /home/solidity/Code/krab/krab_test.go:62 +0x4ac
testing.tRunner(0xc000152100, 0xc67f38)
        /usr/local/go/src/testing/testing.go:865 +0x164
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:916 +0x65b
FAIL    github.com/RTradeLtd/krab       0.166s
```

--- a/datastore.go
+++ b/datastore.go
@@ -94,6 +94,22 @@ func (d *Datastore) Query(q query.Query) (query.Results, error) {
 	resBuilder := query.NewResultBuilder(q)
 	if err := d.db.View(func(tx *bbolt.Tx) error {
 		cursor := tx.Bucket(d.bucket).Cursor()
+		if q.Prefix == "" {
+			for k, _ := cursor.First(); k != nil; k, _ = cursor.Next() {
+				result := query.Result{}
+				key := datastore.NewKey(fmt.Sprintf("%v", k))
+				result.Entry.Key = key.String()
+				if !q.KeysOnly {
+					result.Entry.Value, result.Error = d.Get(key)
+				}
+				select {
+				case resBuilder.Output <- result:
+				default:
+					continue
+				}
+			}
+			return nil
+		}
 		pref := []byte(q.Prefix)
 		for k, _ := cursor.Seek(pref); bytes.HasPrefix(k, pref); k, _ = cursor.Next() {
 			result := query.Result{}

--- a/datastore.go
+++ b/datastore.go
@@ -1,7 +1,6 @@
 package dsbbolt
 
 import (
-	"fmt"
 	"os"
 
 	"bytes"
@@ -97,7 +96,7 @@ func (d *Datastore) Query(q query.Query) (query.Results, error) {
 		if q.Prefix == "" {
 			for k, _ := cursor.First(); k != nil; k, _ = cursor.Next() {
 				result := query.Result{}
-				key := datastore.NewKey(fmt.Sprintf("%v", k))
+				key := datastore.NewKey(string(k))
 				result.Entry.Key = key.String()
 				if !q.KeysOnly {
 					result.Entry.Value, result.Error = d.Get(key)
@@ -113,7 +112,7 @@ func (d *Datastore) Query(q query.Query) (query.Results, error) {
 		pref := []byte(q.Prefix)
 		for k, _ := cursor.Seek(pref); bytes.HasPrefix(k, pref); k, _ = cursor.Next() {
 			result := query.Result{}
-			key := datastore.NewKey(fmt.Sprintf("%v", k))
+			key := datastore.NewKey(string(k))
 			result.Entry.Key = key.String()
 			if !q.KeysOnly {
 				result.Entry.Value, result.Error = d.Get(key)

--- a/datastore_test.go
+++ b/datastore_test.go
@@ -62,12 +62,24 @@ func Test_Datastore(t *testing.T) {
 	} else if size != len([]byte("hello world")) {
 		t.Fatal("incorrect data size")
 	}
-
+	// test a query where we specify a search key
 	rs, err := ds.Query(query.Query{Prefix: key.String()})
 	if err != nil {
 		t.Fatal(err)
 	}
 	res, err := rs.Rest()
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, v := range res {
+		fmt.Printf("%+v\n", v)
+	}
+	// test a query where we dont specify a search key
+	rs, err = ds.Query(query.Query{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	res, err = rs.Rest()
 	if err != nil {
 		t.Fatal(err)
 	}

